spring-data-cassandra/src/main/java/org/springframework/data/cassandra/core/CassandraTemplate.java

```diff
@@ -202,6 +214,15 @@ public CqlOperations getCqlOperations() {
 		return getMappingContext().getRequiredPersistentEntity(ClassUtils.getUserClass(entityType));
 	}
 
+	@org.springframework.lang.Nullable
+	private String guessTableName(Object entity) {
```

Comment:
How about reusing `getTableName(…)`?

Reply:
At first I have reused `#getTableName(...)`. The only issue is that `CustomConversionIntegrationTests` fails. `#getTableName(...)` raises exception when provided Java type is not an entity. That is the reason of introducing the `#guessTableName(...)` method and making table name optional in the event POJOs.

Reply:
When `CustomConversionIntegrationTests` starts failing, then we need to investigate the actual cause and how to fix it. Happy to support you there.